packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMesh.h

```diff
@@ -0,0 +1,366 @@
+
+  /**
+   * Return the coordinates for all nodes on this processor.  The
+   * first 3 entries in the 'coord' vector are the x, y, and z
+   * coordinates of the first node, etc.  The 'coord' vector will be
+   * resized to the size required to contain the nodal coordinates;
+   * all information in 'coord' will be overwritten.
+   */
+  virtual void coordinates(std::vector<double> &coord) const;
+  virtual void coordinates(double *coord) const;
+
+  /**
+   * Return the coordinates for all nodes on this processor in
+   * separate vectors. The vectors will be resized to the size
+   * required to contain the nodal coordinates; all information in
+   * the vectors will be overwritten.
+   */
+  virtual void coordinates(std::vector<double> &x, std::vector<double> &y, std::vector<double> &z) const;
+
+  /**
+   * Return the coordinates for componenet 'comp' (1=x, 2=y, 3=z)
+   * for all nodes on this processor. The
+   * vector will be resized to the size required to contain the
+   * nodal coordinates; all information in the vector will be
+   * overwritten.
+   * It is an error to request the coordinates via this function
+   * if a rotation is defined.
+   */
+  virtual void coordinates(int component, std::vector<double> &xyz) const;
+  virtual void coordinates(int component, double *xyz) const;
+
+  size_t get_variable_count(Ioss::EntityType type) const
+  {
+    return m_variableCount.find(type) != m_variableCount.end() ? m_variableCount.find(type)->second : 0;
+  }
+
+  std::vector<std::string> get_part_names() const;
+  int64_t get_part_id(const std::string &name) const;
+
+  unsigned spatial_dimension() const;
+
+ private:
+  template <typename INT>
+  void raw_element_map(int64_t block_number, std::vector<INT> &map) const;
+  template <typename INT>
+  void raw_element_map(std::vector<INT> &map) const;
+  template <typename INT>
+  void raw_connectivity(int64_t block_number, INT *connect) const;
+  template <typename INT>
+  void raw_node_map(std::vector<INT> &map) const;
+
+  void set_variable_count(const std::string &type, size_t count);
+  void parse_options(const std::vector<std::string> &groups);
+  void parse_coordinates_option(const std::vector<std::string> &option);
+  void parse_dimension_option(const std::vector<std::string> &option);
+
+  void initialize();
+
+  void build_part_to_topology_map();
+  void build_block_partition_map();
+  void build_element_connectivity_map();
+
+  std::vector<int64_t> get_part_ids(const std::vector<std::string> &partNames);
+  std::vector<size_t> get_part_offsets(const std::vector<int64_t> &partIds);
+
+  Topology get_topology_for_part(int64_t id) const;
+
+  std::set<int64_t> get_local_element_ids_for_block(int64_t id) const;
+
+  unsigned m_dimension{3};
+  size_t m_processorCount{0};
+  size_t m_myProcessor{0};
+
+  size_t m_timestepCount{0};
+  std::map<Ioss::EntityType, size_t> m_variableCount;
+
+  std::vector<double> m_rawCoordinates;
```

Comment:
It looks like this is populated during parsing.  It is then used to fill the vectors in `m_coordinates` at which time we have the same/similar data in both locations.  After filling, it looks like the only use of `m_rawCoordinates` is to check whether it is empty.  I realize that text meshes are usually "small", but seems like a waste to keep similar data in two locations.  Maybe I am missing the need for both.

Reply:
This is correct ... I was actually thinking of optimizing storage by not keeping it around longer that necessary. Let me see what I can do about making it a temporary object and I'll upload a new commit

Reply:
A thought occured to me that if we were to formalize the syntax of TextMesh such that the parsing of coordinates/sidesets/nodesets/etc was done natively in TextMesh, then there wouldn't be a need for this container for the raw coordinates.

Reply:
Yes, that sounds right.  I think that would be good...